dist_run.py

```diff
@@ -6,67 +6,151 @@
 
 # Run command:
 # torchrun --nproc-per-node 4 dist_run.py
-
 import torch
 import torch.distributed as dist
 from torch.distributed.pipelining import PipelineStage, ScheduleGPipe
-
-from build.model import ModelArgs
+from build.model import TransformerArgs
 from build.model_dist import TransformerStage
+from distributed.logging_utils import setup_logging
+from distributed.safetensor_utils import (
+    get_hf_config_file,
+    get_hf_weight_map_and_path,
+    load_safetensor_weights,
+)
+from distributed.dtensor_utils import find_cpu_tensors, record_module_dtypes
+from distributed.utils import Color as color
+
+MODEL_NAME = "Transformer-2-7b-chat-hf"
+NAME_TO_HF_MODEL_ID_AND_DTYPE = {
+    "Transformer-2-7b-chat-hf": ("meta-llama/Llama-2-7b-chat-hf", torch.float16),
+}
+
+
+def _init_distributed():
+    dist.init_process_group("nccl")
+    rank = dist.get_rank()
+    world_size = dist.get_world_size()
+    torch.cuda.set_device(rank)
+    return rank, world_size
+
+
+def _create_device_mesh(mesh_dimensions):
+    return dist.init_device_mesh("cuda", mesh_dimensions, mesh_dim_names=("pp", "tp"))
+
+
+def _load_model_weights(stage_module, hf_model_name, device, logger):
+    weight_map, weight_path, key_map = get_hf_weight_map_and_path(hf_model_name)
+    num_loaded_weights, num_missing_weights = load_safetensor_weights(
+        stage_module, weight_map, weight_path, key_map, device
+    )
+    logger.info(
+        f"Success - Loaded {num_loaded_weights} weights, {num_missing_weights} missing weights"
+    )
+    if num_missing_weights > 0:
+        raise ValueError(f"Missing {num_missing_weights} weights")
+
+
+def _cleanup():
+    dist.barrier()
```

Comment:
nit: how about commenting out the barrier, and add a comment saying that "Uncomment it if you hit a hang"?